apps/meteor/client/views/root/hooks/useRoomMessageUsers.ts

```diff
@@ -0,0 +1,77 @@
+import { useUser, useUserId } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import { useCallback, useMemo } from 'react';
+
+import { Messages } from '../../../../app/models/client';
+import { roomMessageUsersQueryKeys } from '../../../lib/queryKeys';
+import { useRoom } from '../../room/contexts/RoomContext';
+
+export type RoomMessageUser = {
+	_id: string;
+	username: string;
+	name?: string;
+	ts: Date;
+	suggestion?: boolean;
+};
+
+export const useRoomMessageUsers = () => {
+	const uid = useUserId();
+	const user = useUser();
+	const rid = useRoom()._id;
+
+	const queryParams = useMemo(
+		() => ({
+			rid,
+			username: user?.username,
+		}),
+		[rid, user?.username],
+	);
+
+	const fetchRoomMessageUsers = useCallback(async (): Promise<RoomMessageUser[]> => {
+		if (!queryParams.rid || !queryParams.username) {
+			return [];
+		}
+
+		const messages = Messages.find(
+			{
+				rid,
+				'u.username': { $ne: queryParams.username },
+				't': { $exists: false },
+				'ts': { $exists: true },
```

Comment:
This filter was not on the other Query. Can you explain why did you add it here?

Reply:
I did this so I don't need to do a verification like this one on the `ComposerPopupProvider.tsx`:
```
.filter((u) => !!u.ts && (filterRegex ? u.username.match(filterRegex) || u.name?.match(filterRegex) : true))
```

This way the overall code is more readable and the array functions are shorter.